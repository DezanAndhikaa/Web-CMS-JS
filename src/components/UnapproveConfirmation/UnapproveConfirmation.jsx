import React from 'react';
import { DialogContent, Button, Modal, CircularProgress } from '@material-ui/core';
import { ImgSendtoEdit, ImgCancelApprove, ImgCancelEditSucc } from '../../assets/imgs';
import './UnapproveConfirmation.scss';
import CloseNotif from '../CloseNotif/CloseNotif';
import { ApiRequestActionsStatus } from '../../core/RestClientHelpers';
import { Menu } from '../../constants';
import SapGuide from '../../views/planning/details/components/SapIssuePages/private-components/SapGuide/SapGuide';

export default class UnapproveConfirmation extends React.PureComponent {

  state={
    isShowModalUnapprove: false,
    isShowModalSend: false,
    isShowSapGuide: false
  }

  componentDidUpdate = (prevProps) =>{
    if (prevProps.openModal !== this.props.openModal)
    this.setState({
      isShowModalUnapprove: this.props.openModal
    })
  }

  isClickedSend = () => {
    this.setState({
      isShowModalUnapprove: !this.state.isShowModalUnapprove,
      isShowModalSend: !this.state.isShowModalSend
    })
  }

  isClosedSend = () => {
    this.setState({
      isShowModalSend: !this.state.isShowModalSend,
      isShowModalUnapprove: !this.state.isShowModalUnapprove
    })
  }

  isShowGuide = () => {
    this.setState({
      isShowModalUnapprove: !this.state.isShowModalUnapprove,
      isShowSapGuide: !this.state.isShowSapGuide
    })
  }

  _renderSendtoEdit(){
    return(
      <Modal open={this.props.openModal} className="modal-unapprove">
        <DialogContent className="unapprove-confirmation-content">
          <div className="confirmation-modal-unapprove">
            <CloseNotif onClose={this.props.onClose}/>
            <div className="confirmation-container-unapprove">
              <p className="confirmation-title-unapprove">Send to Edit</p>
              <p className="confirmation-title-unapprove">Lifetime Component</p>
              <img className="confirmation-image-unapprove" src={ImgSendtoEdit} alt="" />
              <p className="confirmation-caption-unapprove">Are you sure want to Not Approve <b>{this.props.totalData} items?</b></p>
              <div className="btn-row">
                <Button className="button-yes" onClick={this.props.onSendtoEdit}>Yes</Button> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                <Button className="button-no" onClick={this.props.onClose}>No</Button>
              </div>
            </div>
          </div>
        </DialogContent>
      </Modal>
    )
  }

  _renderSapGuide(){
    return(
      <SapGuide 
        {...this.props}
        {...this.state}
        openModal= {this.state.isShowSapGuide}
      />
    )
  }

  renderCircularProgress() {
    return <CircularProgress size={100} className="circular-progress" />;
  }

  handleClick = (menu, tab) => {
    this.props.push({
      pathname: menu,
      whichTab: tab
    });
  }

	render() {
    if(this.props.idConfirm === "Cancel"){
        return (
          <>
           {this.state.isShowModalUnapprove && (
            <Modal open={this.state.isShowModalUnapprove} className="modal-unapprove">
            <DialogContent className="unapprove-confirmation-content">
              <div className="confirmation-modal-unapprove">
                <CloseNotif onClose={this.props.onClose}/>
                <div className="confirmation-container-unapprove">
                  <p className="confirmation-title-unapprove">Cancel Approve</p>
                  {this.props.idCancel === "Sales" ? <p className="confirmation-title-unapprove">Sales Order</p> :
                  <p className="confirmation-title-unapprove">Service Order</p>}
                  <img className="confirmation-image-unapprove" src={ImgCancelApprove} alt="" />
                  <p className="confirmation-caption-unapprove"><b>Select one</b> to continue cancel approve</p>
                  <div className="btn-row">
                    {this.props.whichTabs ? <Button className="button-edit-lt" onClick={() => this.isClickedSend()}>Edit Lifetime</Button> : null }
                    <Button className={this.props.whichTabs ? "button-sap-issue" : "button-sap-issue-service"} onClick={this.props.whichTabs? () => this.handleClick(Menu.PLANNING_SAP, 'sales') : this.handleClick(Menu.PLANNING_SAP, 'service')}>SAP Issue</Button>
                  </div>
                  <div className={this.props.whichTabs ? "labelMax" : "labelMax-service" }>
                    <label>* Max 5 Items</label>
                  </div>
                </div>
              </div>
            </DialogContent>
          </Modal>
          )     
          }
          {this.props.fetchStatusPutSAPIssue === ApiRequestActionsStatus.LOADING &&  (
            this.renderCircularProgress()
          )}
          {this.state.isShowModalSend && (this._renderSendtoEdit() )}
          {this.state.isShowSapGuide && (this._renderSapGuide() )}
        </>
        );
    } else if(this.props.idConfirm === "Send Success"){
      return(
        <Modal open={this.props.openModal} className="modal-unapprove">
        <DialogContent className="unapprove-confirmation-content">
          <div className="confirmation-modal-unapprove">
            <div className="confirmation-container-unapprove">
              <p className="title-success-unapprove">Successful</p>
              <img className="confirmation-success-unapprove" src={ImgCancelEditSucc} alt="" />
              <p className="confirmation-caption-unapprove">You have sent data to be repaired again</p>
              <div className="btn-row">
                <Button className="button-continue" onClick={ () => {this.props.onClose()} }>Continue</Button>
              </div>
            </div>
          </div>
        </DialogContent> 
      </Modal>
      )
    }
	}
}