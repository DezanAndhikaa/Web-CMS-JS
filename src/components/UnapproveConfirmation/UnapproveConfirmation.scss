@import '../../themes/functions';
@import '../../themes/variables';

.modal-unapprove{
    width: fit-content;
    height: fit-content;
    margin: auto;
    .unapprove-confirmation-content{
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 0;
        .confirmation-modal-unapprove{
            width: vw(519);
            // height: vh(481);
            background-color: white;
            margin: auto;
            border-radius: vw(15);
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            padding: vw(15) vw(24) vw(30) vw(24);
            .confirmation-container-unapprove{
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: center;
                flex: 1;
                .confirmation-title-unapprove{
                    line-height: vw(20);
                    margin: 0;
                    padding-top: vw(20);
                    font-size: vw(28);
                    font-family: 'SFUIDisplay-Bold';
                    color: #3C3B3B;
                }
                .confirmation-image-unapprove{
                    width: vw(175);
                    margin-top: vw(16);
                    margin-bottom: vw(25);
                }
                .title-success-unapprove{
                    line-height: vw(20);
                    margin-top: vw(38);
                    font-size: vw(28);
                    font-family: 'SFUIDisplay-Bold';
                    color: #3C3B3B;
                }
                .confirmation-success-unapprove{
                    margin-top: vw(30);
                    margin-bottom: vw(35);
                }
                .confirmation-caption-unapprove{
                    margin: 0;
                    font-size: vw(20);
                    font-family: 'SFUIDisplay-Regular';
                    opacity: 0.5;
                    text-align: center;
                    color: rgb(0, 0, 0);
                }
            }
        }
    }
    .btn-row{
        flex: 1;
        display: flex;
        flex-direction: row;
        margin-top: vw(20);
        width: 100%;
        .button-edit-lt{
            margin-right: 15px;
            display: flex;
            align-items: center;
            justify-content: center;
            text-transform: capitalize;
            width: 50%; //vw(100.22);
            height: vw(45.07);
            font-size: vw(13) !important;
            font-weight: bold;
            line-height: vw(16) !important;
            font-family: 'Lato-Regular';
            background-color: #F2F2F4;
            color: #1D1D1D;
            border: 1px solid map-get($colors, primary);
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            cursor: pointer;
        }
        .button-edit-lt:hover {
            background-color: map-get($colors, primary);
            color: white;
        }
        .button-sap-issue{
            margin-left: 15px;
            display: flex;
            align-items: center;
            justify-content: center;
            text-transform: capitalize;
            width: 50%; //vw(170.22);
            height: vw(45.07);
            font-size: vw(13) !important;
            font-weight: bold;
            line-height: vw(16) !important;
            font-family: 'Lato-Regular';
            background-color: #F2F2F4;
            color: #1D1D1D;
            border: 1px solid #FC9517;
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            cursor: pointer;
        }
        .button-sap-issue:hover {
            background-color: #FC9517;
            color: white;
        }
        .button-yes{
            display: flex;
            align-items: center;
            justify-content: center;
            text-transform: capitalize;
            width: 40%; //vw(100.22);
            height: vw(45.07);
            font-size: vw(13) !important;
            font-weight: bold;
            line-height: vw(16) !important;
            font-family: 'Lato-Regular';
            background-color: #F2F2F4;
            color: #1D1D1D;
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            cursor: pointer;
        }
        .button-yes:hover {
            background-color: rgb(227, 227, 230);
        }
        .button-no{
            display: flex;
            align-items: center;
            justify-content: center;
            text-transform: capitalize;
            width: 60%; //vw(170.22);
            height: vw(45.07);
            font-size: vw(13) !important;
            font-weight: bold;
            line-height: vw(16) !important;
            font-family: 'Lato-Regular';
            background-color: map-get($colors , primary);
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            cursor: pointer;
        }
        .button-no:hover {
            background-color: map-get($colors, primary-dark);
        }
        .button-continue{
            display: flex;
            align-items: center;
            justify-content: center;
            text-transform: capitalize;
            width: 100%; //vw(170.22);
            height: vw(45.07);
            font-size: vw(13) !important;
            font-weight: bold;
            line-height: vw(16) !important;
            font-family: 'Lato-Regular';
            background-color: map-get($colors , primary);
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            cursor: pointer;
        }
    }
    .labelMax{
        margin-top: 1%;
        margin-left:50%;
        font-size: vw(13) !important;
        font-weight: bold;
        line-height: vw(16) !important;
        font-family: 'Lato-Regular';
        color: red;
    }
}

@media (min-width: 600px) and (max-width: 1025px) {    
    .modal-unapprove{
        width: fit-content;
        height: fit-content;
        margin: auto;
        .unapprove-confirmation-content{
            display: flex;
            justify-content: center;
            align-items: center;
            padding: 0;
            .confirmation-modal-unapprove{
                width: vw-tab(450);
                height: vw-tab(300);
                background-color: white;
                margin: auto;
                border-radius: vw-tab(5);
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: space-between;
                padding: vw-tab(15) vw-tab(24) vw-tab(30) vw-tab(24);
                .confirmation-container-unapprove{
                    display: flex;
                    flex-direction: column;
                    align-items: center;
                    justify-content: center;
                    flex: 1;   
                    .confirmation-title-unapprove{
                        margin: 0;
                        font-size: vw-tab(33);
                        font-family: 'SFUIDisplay-Bold';
                        color: #3C3B3B;
                    }
                    .confirmation-image-unapprove{
                        width: vw-tab(150);
                        margin-top: vw-tab(16);
                        margin-bottom: vw-tab(25);
                    }
                    .confirmation-caption-unapprove{
                        margin: 0;
                        font-size: vw-tab(20);
                        font-family: 'SFUIDisplay-Regular';
                        text-align: center;
                        opacity: 0.5;
                        color: #3C3B3B;
                    }
                }
            }
        }
    }
}

@media (max-width: 599px) {    
    .modal-unapprove{
        width: fit-content;
        height: fit-content;
        margin: auto;
        .unapprove-confirmation-content{
            display: flex;
            justify-content: center;
            align-items: center;
            padding: 0;
            .confirmation-modal-unapprove{
                margin: 0;
                width: vw-mobile(215);
                height: vw-mobile(200);
                background-color: white;
                border-radius: vw-mobile(15);
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: space-between;
                padding: vw-mobile(10);
                .confirmation-container-unapprove{
                    display: flex;
                    flex-direction: column;
                    align-items: center;
                    justify-content: center;
                    flex: 1;   
                    .confirmation-title-unapprove{
                        margin: 0;
                        font-size: vw-mobile(20);
                        font-family: 'SFUIDisplay-Bold';
                        color: #3C3B3B;
                    }
                    .confirmation-image-unapprove{
                        width: vw-mobile(100);
                        margin-top: vw-mobile(5);
                        margin-bottom: vw-mobile(10);
                    }
                    .confirmation-caption-unapprove{
                        margin: 0;
                        font-size: vw-mobile(15);
                        font-family: 'SFUIDisplay-Regular';
                        text-align: center;
                        opacity: 0.5;
                        color: #3C3B3B;
                    }
                }
            }
        }
    }
}