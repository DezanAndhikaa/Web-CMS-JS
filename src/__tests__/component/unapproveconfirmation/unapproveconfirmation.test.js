import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import UnapproveConfirmation from '../../../components/UnapproveConfirmation/UnapproveConfirmation';

Enzyme.configure({adapter:new Adapter()});
it("render modal unapprove confirmation correctly", () => {
    const wrapper = shallow (
        <UnapproveConfirmation />
    )
    expect(wrapper).toMatchSnapshot();
})