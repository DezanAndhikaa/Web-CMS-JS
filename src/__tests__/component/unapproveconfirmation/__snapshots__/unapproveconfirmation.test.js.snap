// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render input text correctly 1`] = `ShallowWrapper {}`;
